File: Proc-0/Text/CharSeeker.cs
/// <summary>
/// Интерфейс поиска в символьном потоке.
/// </summary>
interface CharSeeker
{
	/// <summary>
	/// Номер текущего символа.
	/// </summary>
	long Position { get; protected set; }

	/// <summary>
	/// Номер текущего символа в строке.
	/// </summary>
	long Character { get; protected set; }

	/// <summary>
	/// Номер позиции символа в строке.
	/// Табуляция занимает несколько позиций.
	/// </summary>
	long Column { get; protected set; }

	/// <summary>
	/// Номер текущей строки.
	/// </summary>
	long Line { get; protected set; }

	bool SeekChar(long offset);
}

File: Proc-0/src/main/kotlin/Text/CharSeeker.kt
package proc.Text

/**
 * Интерфейс поиска в символьном потоке.
 */
interface CharSeeker {

	/**
	 * Номер текущего символа.
	 */
	var Position: Long
		get
		set // protected

	/**
	 * Номер текущего символа в строке.
	 */
	var Character: Long
		get
		set // protected

	/**
	 * Номер позиции символа в строке.
	 * Табуляция занимает несколько позиций.
	 */
	var Column: Long
		get
		set // protected

	/**
	 * Номер текущей строки.
	 */
	var Line: Long
		get
		set // protected

	fun SeekChar(offset: Long): Boolean
}
